hal/src/stm32f2xx/i2c_hal.c

```diff
@@ -1026,6 +1026,8 @@ static void HAL_I2C_EV_InterruptHandler(HAL_I2C_Interface i2c)
             // the device will continue pulling SCL low. To avoid that, disable clock stretching
             // when the tx buffer is exhausted to release SCL.
             I2C_StretchClockCmd(i2cMap[i2c]->I2C_Peripheral, DISABLE);
+            HAL_Delay_Microseconds(1);
```

Comment:
We shouldn't delay in an ISR. Would a simple `__DMB()` or `__DSB()` + `__ISB()` work here?

Reply:
Updated.